pkgs/development/python-modules/pysaml2/default.nix

```diff
@@ -1,21 +1,32 @@
 { stdenv
 , buildPythonPackage
-, fetchPypi
+, fetchFromGitHub
 , cryptography, defusedxml, future, pyopenssl, dateutil, pytz, requests, six
+, mock, pyasn1, pymongo, pytest, responses, xmlsec
 }:
 
 buildPythonPackage rec {
   pname = "pysaml2";
   version = "4.6.5";
 
-  src = fetchPypi {
-    inherit pname version;
-    sha256 = "0379d6d91b3573c4b2eff49c2658d472b47849096b6a816a3b67ec2f3054ed71";
+  # No tests in PyPI tarball
+  src = fetchFromGitHub {
+    owner = "IdentityPython";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "0xlbr52vzx1j9sg65jhqv01vp4a49afjy03lc2zb0ggx0xxzngvb";
   };
 
   propagatedBuildInputs = [ cryptography defusedxml future pyopenssl dateutil pytz requests six ];
 
-  doCheck = false;
+  checkInputs = [ mock pyasn1 pymongo pytest responses xmlsec ];
```

Comment:
Shouldn't xmlsec be a buildInput ? According to a dev it seems to be used at runtime.

https://github.com/IdentityPython/pysaml2/issues/474#issuecomment-343232351